lib/fhir_models/fhir.rb

```diff
@@ -16,11 +16,15 @@ def self.default_logger
 
   def self.from_contents(contents)
     doc = Nokogiri::XML(contents)
-    if doc.errors.empty?
-      FHIR::Xml.from_xml(contents)
-    else
-      FHIR::Json.from_json(contents)
-    end
+    resource =
+      if doc.errors.empty?
+        FHIR::Xml.from_xml(contents)
+      else
+        FHIR::Json.from_json(contents)
+      end
```

Comment:
I do not know if these are part of the 'public' API.  If they were and someone used them they wouldn't have this info saved... which is unintuitive.

They aren't mentioned in the README, so maybe it's ok? Depends on how defensive we want this update to be...

Reply:
I don't consider them part of the public api, but it's easy enough to move the saving there.